pwa/src/templates/templateParts/mappingForm/MappingFormTemplate.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from "react";
+import * as styles from "./MappingFormTemplate.module.css";
+import { useForm } from "react-hook-form";
+import FormField, { FormFieldInput, FormFieldLabel } from "@gemeente-denhaag/form-field";
+import { useTranslation } from "react-i18next";
+import { InputText } from "@conduction/components";
+import { useQueryClient } from "react-query";
+import { IsLoadingContext } from "../../../context/isLoading";
+import { useMapping } from "../../../hooks/mapping";
+import { CreateKeyValue, InputCheckbox, Textarea } from "@conduction/components/lib/components/formFields";
+
+interface MappingFormTemplateProps {
+  mapping?: any;
+}
+
+export const formId: string = "mapping-form";
+
+export const MappingFormTemplate: React.FC<MappingFormTemplateProps> = ({ mapping }) => {
+  const { t } = useTranslation();
+  const [isLoading, setIsLoading] = React.useContext(IsLoadingContext);
+
+  const [_mapping, setMapping] = React.useState<any[]>([]);
+  const [unset, setUnset] = React.useState<any[]>([]);
+  const [cast, setCast] = React.useState<any[]>([]);
+
+  const queryClient = useQueryClient();
+  const createOrEditMapping = useMapping(queryClient).createOrEdit(mapping?.id);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    control,
+    formState: { errors },
+  } = useForm();
+
+  const onSubmit = (data: any): void => {
+    createOrEditMapping.mutate({ payload: data, id: mapping?.id });
+  };
+
+  const handleSetFormValues = (): void => {
+    const basicFields: string[] = ["name", "reference", "version", "description", "passTrough"];
+    basicFields.forEach((field) => setValue(field, mapping[field]));
+
+    if (Array.isArray(mapping.mapping) || mapping.mapping === undefined) {
+      setMapping(mapping.mapping);
+    } else {
+      const newMapping = Object.entries(mapping.mapping).map(([key, value]) => ({ key, value: value }));
+      setMapping(newMapping);
+    }
```

Comment:
I think the gateway should always be returning this value as an array. If they do, the code could be:
```
mapping.mapping && setDefaultMappings(Object.entries(mapping.mapping).map(([key, value]) => ({ key, value })));
```